idle-crafting.lua

```diff
@@ -6,6 +6,55 @@ local widgets = require('gui.widgets')
 local repeatutil = require("repeat-util")
 local orders = require('plugins.orders')
 
+---iterate over input materials of workshop with stockpile links
+---@param workshop df.building_workshopst
+---@param action fun(item:df.item):any
+local function for_inputs(workshop, action)
+    if #workshop.profile.links.take_from_pile == 0 then
+        dfhack.error('workshop has no links')
+    else
+        for _, stockpile in ipairs(workshop.profile.links.take_from_pile) do
+            for _, item in ipairs(dfhack.buildings.getStockpileContents(stockpile)) do
+                if(item:isAssignedToThisStockpile(stockpile.id)) then
+                    for _, contained_item in ipairs(dfhack.items.getContainedItems(item)) do
+                       action(contained_item)
+                    end
+                else
+                    action(item)
+                end
+            end
+        end
+        for _, contained_item in ipairs(workshop.contained_items) do
+            if contained_item.use_mode == 0 then
```

Comment:
please use the enum value instead of a magic 0

I believe it's `df.building_item_role_type.TEMP`